map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java

```diff
@@ -86,48 +87,45 @@ public void start(Stage stage) {
 
       // create a label to display the view status
-      Label layerViewStatusLabel = new Label("Current view status: ");
+      var layerViewStatusLabel = new Label("Current view status: ");
 
       // create a checkbox UI that toggles the visibility of the feature layer
-      CheckBox visibilityCheckBox = new CheckBox();
-      visibilityCheckBox.setText("Layer visible");
-      visibilityCheckBox.setSelected(true);
+      var visibilityCheckBox = new CheckBox();
       visibilityCheckBox.setDisable(true);
+      visibilityCheckBox.setSelected(true);
+      visibilityCheckBox.setText("Layer visible");
       visibilityCheckBox.setOnAction(event -> featureLayer.setVisible(visibilityCheckBox.isSelected()));
 
-      // create a button and a listener to load a feature layer
-      Button loadLayerButton = new Button("Load Layer");
+      // create a button and a listener to add the feature layer
+      var loadLayerButton = new Button("Load layer");
       loadLayerButton.setOnAction(event -> {
 
-        // disable the checkbox when loading/reloading the feature layer, and, if it exists, remove it by clearing the map's operational layers
-        if (!visibilityCheckBox.isDisabled()) {
-          visibilityCheckBox.setDisable(true);
-        }
-
+        // clear the map's operational layers
         if (featureLayer != null) {
           map.getOperationalLayers().clear();
         }
 
         // create a new feature layer from a portal item
-        final PortalItem portalItem = new PortalItem(new Portal("https://runtime.maps.arcgis.com/"),
+        final var portalItem = new PortalItem(new Portal("https://runtime.maps.arcgis.com/"),
           "b8f4033069f141729ffb298b7418b653");
         featureLayer = new FeatureLayer(portalItem, 0);
         //  set a minimum and a maximum scale for the visibility of the feature layer
         featureLayer.setMinScale(40000000);
         featureLayer.setMaxScale(4000000);
-        // add the feature layer to the map
+
+        // add the feature layer to the map and update its visibility
         map.getOperationalLayers().add(featureLayer);
+        featureLayer.setVisible(visibilityCheckBox.isSelected());
 
-        // enable UI interactions once the feature layer has loaded
-        featureLayer.addDoneLoadingListener(() -> {
-          if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
-            visibilityCheckBox.setDisable(false);
-            featureLayer.setVisible(visibilityCheckBox.isSelected());
-            loadLayerButton.setText("Reload Layer");
-          } else {
-            new Alert(Alert.AlertType.ERROR, "Feature layer failed to load").show();
-          }
-        });
+        // update the button text when the feature layer has loaded
+        featureLayer.addDoneLoadingListener(() ->
+          loadLayerButton.textProperty().bind(Bindings.createStringBinding(() ->
+            featureLayer.loadStatusProperty().get() == LoadStatus.LOADED ? "Reload layer" : "Load layer",
```

Comment:
I think "Reload layer" might also be appropriate for a failed layer since the in the API you would call `retryLoadAsync()`. Depends how close to the API you want your button titles to be.